### ISSUE TITLE
Add data values to BlockType

### ISSUE THREAD

**Issue by mcpower** [katharosada/botchallenge#14]
[Data values](http://minecraft.gamepedia.com/Data_values#Data) in Minecraft, also known as "damage values", are ways of storing extra data on items (like orientation of blocks, type of wool, crop growth stages and more). Having access to this from the API would open up lots of new inventions, for example, an efficient auto crop harvester.

However, the implementation may be a bit hard. Items with different data values can't be stacked, and are used differently, but internally are represented by the same block ID. Checking whether an item is valid will need some code modification (as it's currently checking whether they are in a dictionary), and there may be many more problems that I can't think of.

**Comment by katharosada** [comment]
Yeah, I've hit this too. I considered adding an optional byte field to the place function, but it's really ugly and I wanted to find a better way to do it. There's a secondary problem, that the list of official Material names (like those used in the API) are not the same as the names inside the game (e.g. LOG == wood) which I would want to fix too.

The best plan I have yet, is to come up with a list of materials which matches how they appear in game, e.g. list blue wool and red wool separately (and have a mapping from this to id+data pairs on the server), but also have some meta materials or material types e.g. ANY_WOOD or ANY_WOOL to represent any kind of wood/wool, so that when you place a block, or you're searching for a type you don't necessarily need to specify that you're looking at or placing Birch Wood or Grey Wool, you just care that it's wood/wool.
